crates/biome_js_analyze/src/lint/nursery/use_exhaustive_switch_cases.rs

```diff
@@ -0,0 +1,314 @@
+            make::js_call_expression(
+                make::js_identifier_expression(make::js_reference_identifier(make::ident("Error")))
+                    .into(),
+                make::js_call_arguments(
+                    make::token(T!['(']),
+                    make::js_call_argument_list(
+                        [AnyJsCallArgument::AnyJsExpression(
+                            AnyJsExpression::AnyJsLiteralExpression(
+                                make::js_string_literal_expression(make::js_string_literal(
+                                    "TODO: Not implemented yet",
+                                ))
+                                .into(),
+                            ),
+                        )],
+                        [],
+                    ),
+                    make::token(T![')']),
+                ),
+            )
+            .build()
+            .into(),
+        )
+        .build();
+
+        let case_list = stmt.cases();
+        let mut clauses = case_list.iter().collect::<Vec<_>>();
+
+        let leading_trivia = case_list
+            .last()
+            .and_then(|case| case.syntax().first_leading_trivia());
+
+        let throw_stmt: AnyJsStatement = make::js_throw_statement(
+            make::token(T![throw]).with_trailing_trivia([(TriviaPieceKind::Whitespace, " ")]),
+            error_expr.clone().into(),
+        )
+        .with_semicolon_token(make::token(T![;]))
+        .build()
+        .into();
+
+        for ty in state {
+            let mut case_token =
+                make::token(T![case]).with_trailing_trivia([(TriviaPieceKind::Whitespace, " ")]);
+
+            if let Some(leading_trivia) = &leading_trivia {
+                case_token = case_token.with_leading_trivia_pieces(leading_trivia.pieces());
+            } else {
+                case_token = case_token.with_leading_trivia([
+                    (TriviaPieceKind::Newline, "\n"),
+                    (TriviaPieceKind::Whitespace, "  "),
+                ]);
+            }
+
+            let clause = AnyJsSwitchClause::JsCaseClause(make::js_case_clause(
+                case_token,
+                type_to_expression(ty)?,
+                make::token(T![:]).with_trailing_trivia([(TriviaPieceKind::Whitespace, " ")]),
+                make::js_statement_list([throw_stmt.clone()]),
+            ));
+
+            clauses.push(clause);
+        }
+
+        mutation.replace_node(case_list, make::js_switch_case_list(clauses));
+
+        let message = markup! { "Add missing cases to the switch statement." };
+
+        Some(JsRuleAction::new(
+            ctx.metadata().action_category(ctx.category(), ctx.group()),
+            ctx.metadata().applicability(),
+            message,
+            mutation,
+        ))
+    }
+}
+
+/// Unwraps [`TypeInner::InstanceOf`] and [`TypeInner::TypeofType`] to get the inner type.
+/// TODO: I am not sure that if this should be in the type flattening logic.
```

Comment:
```suggestion
// TODO: I am not sure that if this should be in the type flattening logic.
```

`TODO` comments should be part of the docs